doc/sphinx/source/project/introduction.rst

```diff
@@ -4,165 +4,191 @@
+approach: it provides an inherently efficient graph-based compilation 
+infrastructure designed to be compatible with many upcoming DL ASICs while 
+also unlocking a massive performance boost on any existing hardware targets 
+in a network, whether they are CPUs, GPUs, or other custom silicon. nGraph 
+provides optimization opportunities at the graph level, where the 
+network-to-device compilation can be managed with a series of "subgraphs"
+that can be handled in either a static or a dynamic manner. With our 
+:doc:`../ops/index` and graph-based infrastructure for neural networks, 
+it's also possible to extract context semantics that make it much easier to 
+work with many of the new and emerging problems in Deep Learning including 
+larger datasets, data that must be encrypted, and data-in-motion. Our solution 
+also addresses the scalability issue with kernel libraries, the current 
+popular solution to accelerating deep learning performance. O
 
 
 Motivations
 ===========
 
-Kernel libraries do not support graph-level optimizations
----------------------------------------------------------
-
-A framework designed for training using GPUs requires integration with a kernel 
-library unique to that vendor's hardware. For example, after integration, a 
-kernel library can run operations that it is "familar" with optimally; however, 
-the graph itself within any larger :term:`NN` won't be optimal.
-
-.. _figure-A:
+The current state-of-the-art software solution for speeding up deep learning 
+computation is to integrate kernel libraries like Intel® Math Kernel Library 
+for Deep Neural Networks (Intel® MKL DNN) and Nvidia\*'s CuDNN into deep 
+learning frameworks. These kernel libraries offer a runtime performance boost 
+on specific hardware targets through highly-optimized kernels and other 
+operator-level optimizations.
 
-.. figure:: ../graphics/framework-to-kernel-lib.png
-   :width: 555px
-   :alt: 
+However, kernel libraries have three main problems: 
 
-   Figure A: Lack of graph-level optimization makes framework-to-kernel library
-   integration enormously inefficient. The computation graph above represents 
-   the computation: "A plus B times C".
+#. Kernel libraries do not support graph level optimizations.
+#. Framework integration of kernel libraries does not scale.
+#. There are too many kernels to write, and they require expert knowledge.
 
+The nGraph Compiler stack is designed to address the first two problems. nGraph 
+applies graph-level optimizations by taking the computational graph from a deep 
+learning framework like TensorFlow* and reconstructing it with the nGraph 
+:abbr:`Intermediate Representation (IR)`. The nGraph IR centralizes computational 
+graphs from various frameworks and provides a unified way to connect backends 
+for targeted hardware. From here, PlaidML or one of the nGraph custom 
+transformers can generate code various forms including LLVM, OpenCL, OpenGL, 
```

Comment:
I'm not sure what a "code various forms" is. Is it code-various forms? or something else.